be/src/exprs/dictmapping_expr.cpp

```diff
@@ -34,17 +34,32 @@ StatusOr<ColumnPtr> DictMappingExpr::evaluate_checked(ExprContext* context, Chun
     if (ptr == nullptr || ptr->is_empty()) {
         return get_child(1)->evaluate_checked(context, ptr);
     }
-    auto target_column = ptr->get_column_by_slot_id(slot_id());
-    auto data_column = ColumnHelper::get_data_column(target_column.get());
 
-    if (data_column->is_binary()) {
-        DCHECK(dict_func_expr == nullptr);
-        return get_child(1)->evaluate_checked(context, ptr);
-    } else if (dict_func_expr != nullptr) {
-        return dict_func_expr->evaluate_checked(context, ptr);
-    } else {
-        return Status::InternalError("unreachable path, dict_func_expr shouldn't be nullptr");
+    if (_children.size() == 2) {
```

Comment:
Explain when children==2 and when children==3.

Reply:
children == 2: DictExpr(string column, string expression)
children == 3: DictExpr(array column, string expression, array expression)
do array-expresion first, then string expression